### PR TITLE
qt 6.7.0

### PR DESCRIPTION
**Pull request by cho-m**
- [ ] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [ ] Have you ensured that your commits follow the [commit style guide](https://docs.brew.sh/Formula-Cookbook#commit)?
- [ ] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [ ] Have you built your formula locally with `HOMEBREW_NO_INSTALL_FROM_API=1 brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [ ] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [ ] Does your build pass `brew audit --strict <formula>` (after doing `HOMEBREW_NO_INSTALL_FROM_API=1 brew install --build-from-source <formula>`)? If this is a new formula, does it pass `brew audit --new <formula>`?

-----

### CODE REVIEW

File: Formula/q/qt.rb [hunk 1]
```diff
@@ -141,18 +130,22 @@ class Qt < Formula
     sha256 "b36a1c245f2d304965eb4e0a82848379241dc04b865afcc4aab16748587e1923"
   end
 
+  # Work around build failure with ninja >= 1.12.0.
+  # Issue ref: https://bugreports.qt.io/browse/QTBUG-124375
+  patch :DATA
```
Comment (cho-m) [inline Formula/q/qt.rb:135]: Hit another issue afterward for `fatal error: 'components/web_cache/public/mojom/web_cache.mojom.h' file not found` at `components/web_cache/browser/web_cache_manager.h`.

Not sure how to resolve this one, so going to use older `ninja` instead.

**Comment by Bo98** [comment]
Yeah, if it's not supported at all in QtWebEngine this should probably be `NOT MACOS AND ...`:

https://github.com/qt/qtwebengine/blob/5c2dcae95a9cb16e2a8f6730304c10397620b7a6/src/core/api/configure.cmake#L165

We can workaround it on our side in the meantime with `-no-feature-webengine-vulkan`

**Comment by paperchalice** [comment]
llvm linkage is related to some Qt tools, like qdoc, update: https://github.com/qt/qttools/blob/dev/configure.cmake

**Comment by cho-m** [comment]
> llvm linkage is related to some Qt tools, like qdoc, lupdate: https://github.com/qt/qttools/blob/dev/configure.cmake `--no-feature-clang --no-feature-clangcpp` might work.

I've added `-no-feature-clang`. I think that should disable `QT_FEATURE_clang` and automatically handled `clangcpp`/`qdoc` features.

File: Formula/q/qt.rb [hunk 2]
```diff
@@ -187,63 +183,73 @@ def install
...
-        -DQT_FEATURE_webengine_system_libxml=ON
-        -DQT_FEATURE_webengine_system_libwebp=ON
-        -DQT_FEATURE_webengine_system_minizip=ON
-        -DQT_FEATURE_webengine_system_opus=ON
-        -DQT_FEATURE_webengine_system_poppler=ON
-        -DQT_FEATURE_webengine_system_pulseaudio=ON
-        -DQT_FEATURE_webengine_system_zlib=ON
+      # NOTE: for QtWebEngine arguments:
+      # * The vendored copy of `libjpeg` is used instead of the brewed copy,
+      #   because the build fails due to a missing symbol otherwise.
```
Comment (cho-m) [inline Formula/q/qt.rb:224]: I will probably try one run with system copy since Arch is using it

File: Formula/q/qt.rb [hunk 3]
```diff
@@ -187,63 +183,73 @@ def install
...
+        -feature-webengine-ozone-x11
+        -system-webengine-alsa
+        -system-webengine-ffmpeg
+        -system-webengine-freetype
+        -system-webengine-harfbuzz
+        -system-webengine-icu
+        -system-webengine-lcms2
+        -system-webengine-libevent
+        -system-webengine-libpng
+        -system-webengine-libvpx
```
Comment (cho-m) [inline Formula/q/qt.rb:240]: And probably remove this. I added it after copying all possible system lib, but noticed Arch ref https://codereview.qt-project.org/c/qt/qtwebengine/+/454908

File: Formula/q/qt.rb [hunk 4]
```diff
@@ -187,63 +183,73 @@ def install
...
       # Explicitly specify QT_BUILD_INTERNALS_RELOCATABLE_INSTALL_PREFIX so
       # that cmake does not think $HOMEBREW_PREFIX/lib is the install prefix.
       cmake_args << "-DQT_BUILD_INTERNALS_RELOCATABLE_INSTALL_PREFIX=#{prefix}"
 
-      # The vendored copy of libjpeg is also used instead of the brewed copy, because the build
-      # fails due to a missing symbol otherwise.
-      # On macOS chromium will always use bundled copies and the QT_FEATURE_webengine_system_*
-      # arguments are ignored.
-      cmake_args += %w[
-        -DQT_FEATURE_webengine_system_alsa=ON
```
Comment (cho-m) [inline Formula/q/qt.rb:229]: Going to switch back to CMake cache variables as there is smart auto-ignore of configure args.

File: Formula/q/qt.rb [hunk 5]
```diff
@@ -187,61 +184,90 @@ def install
...
     ]
 
-    if OS.mac?
-      # Fix a regression in Qt 6.5.2 w.r.t. system libpng
-      # https://bugreports.qt.io/browse/QTBUG-115357
-      cmake_args << "-DQT_FEATURE_webengine_system_libpng=OFF"
-
-      cmake_args << "-DCMAKE_OSX_DEPLOYMENT_TARGET=#{MacOS.version}.0"
+    cmake_args += if OS.mac?
       config_args << "-sysroot" << MacOS.sdk_path.to_s
```
Comment (cho-m) [inline Formula/q/qt.rb:223]: If I need to re-run, then can remove this
```suggestion
```
```
CMake Warning at qtbase/cmake/QtProcessConfigureArgs.cmake:341 (message):
  The -sysroot option is deprecated and no longer has any effect.  It is
  recommended to use a toolchain file instead, i.e.,
  -DCMAKE_TOOLCHAIN_FILE=<filename>.  Alternatively, you may use
  -DCMAKE_SYSROOT option to pass the sysroot to CMake.
```

**Review by carlocab — APPROVED** [review]

**Comment by cho-m** [comment]
ABI changes impacting `pyside` and `pyqt`. `pyside` has 6.7.0 so can upgrade to that.

`pyqt` is more troublesome as they haven't released a new version yet. Changelog references Qt 6.7.0 https://www.riverbankcomputing.com/static/Downloads/PyQt6/ChangeLog-6.7.0.dev2404171245.

I don't see any upstream git (or alternative) repo so may need to generate a patch off upstream's development release. Arch Linux had used a dev release (https://archlinux.org/packages/extra/x86_64/python-pyqt6/) for PyQt but the extra bindings (resources) are still on 6.6.0 so may only need to handle main package.

**Review by chenrui333 — APPROVED** [review]

**Review by fxcoudert — APPROVED** [review]

**Pull request merged**